File: samples/CSharp/Northwind/Territory.cs
using System;
using System.Collections.ObjectModel;
using DbExtensions;

namespace Samples.CSharp.Northwind {

   [Table(Name = "Territories")]
   public class Territory {

      [Column(IsPrimaryKey = true)]
      public string TerritoryID { get; set; }

      [Column]
      public string TerritoryDescription { get; set; }

      [Column]
      public int RegionID { get; set; }

      [Association(OtherKey = nameof(EmployeeTerritory.TerritoryID))]
      public Collection<EmployeeTerritory> EmployeeTerritories { get; } = new Collection<EmployeeTerritory>();

      [Association(ThisKey = nameof(RegionID), IsForeignKey = true)]
      public Region Region { get; set; }
   }
}

File: samples/VisualBasic/Northwind/Territory.vb
Imports System
Imports System.Collections.ObjectModel
Imports DbExtensions

Namespace Northwind

   <Table(Name:="Territories")>
   Public Class Territory

      <Column>
      Property RegionID As Integer

      <Column>
      Property TerritoryDescription As String

      <Column(IsPrimaryKey:=True)>
      Property TerritoryID As String

      <Association(OtherKey:=NameOf(EmployeeTerritory.TerritoryID))>
      ReadOnly Property EmployeeTerritories As New Collection(Of EmployeeTerritory)

      <Association(ThisKey:=NameOf(RegionID))>
      Property [Region] As Region

   End Class

End Namespace
